lib/internal/main/worker_thread.js

```diff
@@ -161,6 +161,11 @@ port.on('message', (message) => {
       });
       process.argv.splice(1, 0, name);
       evalScript(name, filename);
+    } else if (doEval === 'module') {
+      const { evalModule } = require('internal/process/execution');
+      evalModule(filename).catch((e) => {
+        workerOnGlobalUncaughtException(e, true);
+      });
```

Comment:
It might be nice to have tests that this properly propagates a) uncaught exceptions and b) unresolved TLA to the main thread

Reply:
Good catch, unresolved TLA doesn't fire the `error` event…

Reply:
It shouldn’t (or at least it doesn’t do on the main thread either), but it should exit with exit code `13` :)

Reply:
Oh right! So it's working fine already 😅  I've added tests, PTAL